src/detect-http-header.c

```diff
@@ -212,6 +234,66 @@ static int DetectEngineInspectBufferHttpHeader(
     return DETECT_ENGINE_INSPECT_SIG_NO_MATCH;
 }
 
+/** \internal
+ *  \brief custom inspect function to utilize the cached headers
+ */
+static int DetectEngineInspectBufferHttp2Header(DetectEngineCtx *de_ctx,
```

Comment:
I think this should work the same as `DetectEngineInspectBufferGeneric`.

Reply:
ok trying this